projects/plugins/boost/app/assets/src/js/pages/getting-started/getting-started.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+import { Snackbar } from '@wordpress/components';
+import { initializeConnection, getUpgradeURL } from '$lib/stores/connection';
+import { recordBoostEvent } from '$lib/utils/analytics';
+import { navigate } from '$lib/utils/navigate';
+import { BoostPricingTable } from '$features/boost-pricing-table/boost-pricing-table';
+import ActivateLicense from '$features/activate-license/activate-license';
+import Footer from '$layout/footer/footer';
+import Header from '$layout/header/header';
+import styles from './getting-started.module.scss';
+
+type GettingStartedProps = {
+	userConnected: boolean;
+	pricing: ( typeof Jetpack_Boost )[ 'pricing' ];
+	isPremium: boolean;
+	domain: string;
+};
+
+const GettingStarted: React.FC< GettingStartedProps > = ( {
+	userConnected,
+	pricing,
+	isPremium,
+	domain,
+} ) => {
+	const [ selectedPlan, setSelectedPlan ] = useState< 'free' | 'premium' | false >( false );
+	const [ snackbarMessage, setSnackbarMessage ] = useState< string >( '' );
+
+	if ( false !== selectedPlan ) {
+		initialize( selectedPlan );
+	}
+
+	async function initialize( plan: 'free' | 'premium' ) {
+		try {
+			// Make sure there is a Jetpack connection
+			await initializeConnection();
+
+			// Record this selection. This must be done after the connection is initialized.
+			// Possible Events:
+			// * free_cta_from_getting_started_page_in_plugin
+			// * premium_cta_from_getting_started_page_in_plugin
+			recordBoostEvent( `${ plan }_cta_from_getting_started_page_in_plugin`, {} );
+
+			// Go to the purchase flow if the user doesn't have a premium plan.
+			if ( ! isPremium && plan === 'premium' ) {
+				window.location.href = getUpgradeURL( domain, userConnected );
+			}
+			// Otherwise go to dashboard home.
+			// @todo - fix when react routing
+			// navigate( '/', { replace: true } );
+			navigate( '/' );
+		} catch ( e ) {
+			// Display the error in a snackbar message
+			setSnackbarMessage( e.message || 'Unknown error occurred during the plan selection.' );
+		} finally {
+			setSelectedPlan( false );
+		}
```

Comment:
This is almost an infinite loop, or at least without this `finally` it would be. I think this probably needs to be wrapped in a react hook and the initialize function can be lifted out of the component.

Reply:
I'm not sure what moving the function outside the component would look like because of the state setter functions.

Also - how would it be an infinite loop without the `finally`? Because the setter would always get called?

Reply:
I attempted to resolve this in 36ec873f63149e8a2879a0ee5eff65515348a3b2.